### PR TITLE
Adds league container router, closes #170

### LINKED ISSUE #170: League Container router
Would you accept a PR for a router like this?

### PR DESCRIPTION
**Pull request by sagikazarmark**

### CODE REVIEW

**Comment by sagikazarmark** [comment]
Damn, it requires a high PHP version than PHP 5.3. Wasn't it the plan to make PHP 5.4 as a minimum requirement?

**Comment by henrikbjorn** [comment]
Hmm didnt account for the League Container be +5.4, will have to think a bit about raising the minimum php version.

Also it seems like i need to fix the db tests.

**Comment by Baachi** [comment]
@sagikazarmark You can skep the tests if the php version is lower than 5.4, there is no reason to bump the php version for this library.

**Comment by sagikazarmark** [comment]
@Baachi it is not my call. See #163 (where you agreed with it anyway)

**Comment by Baachi** [comment]
Yes i know and i'm a big fan to bump the version to 5.4. I only wanted to give you a hint that this PR can also be accepted without raise the php version.

**Comment by sagikazarmark** [comment]
Oh, I see, thanks.

**Comment by sagikazarmark** [comment]
PR rebased, it should work fine now. I wrote phpunit tests. As soon as package is split up to separate, smaller packages, I am open to writting spec tests for it. (In fact, routers using third party containers could be split up right now as well, since they are not affected by the planned rewrite)

File: tests/Router/LeagueContainerAwareRouterTest.php [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+
+        $this->assertSame($this->container->get('my.service'), $router->map($envelope));
+    }
+
+    public function testAcceptsContainerServiceAsReceiver()
+    {
+        $envelope = new Envelope(new DefaultMessage('SendNewsletter'));
+
+        $router = new LeagueContainerAwareRouter($this->container, array(
+            'SendNewsletter' => 'my.service',
```
Comment (henrikbjorn) [inline tests/Router/LeagueContainerAwareRouterTest.php]: This seems to be the same as the test above, it should have added the mapping after initiating the router right?

Comment (sagikazarmark) [inline tests/Router/LeagueContainerAwareRouterTest.php]: Actually, I copied the Symfony Container test and rewrote everything to League Container. So you might be right. ;)

**Comment by henrikbjorn** [comment]
Thanks for your work on this.

**Pull request merged**